optimade/adapters/base.py

```diff
@@ -0,0 +1,75 @@
+from typing import Union, Dict, Callable, Any
+
+from optimade.models import EntryResource
+
+
+class EntryAdapter:
+    """Base class for lazy resource entry adapters
+    :param entry: JSON OPTIMADE single resource entry.
+    """
+
+    ENTRY_RESOURCE: EntryResource = EntryResource
+    _type_converters: Dict[str, Callable] = {}
```

Comment:
I think this may very easily be done by patching through to the underlying `pydantic` model? But I don't know if that's desireable as such, since we don't have direct control over what is and isn't included that way.

Reply:
In the end, this would simply return the original input, wouldn't it?

Reply:
It would, and I think that would be useful as the input is easily lost otherwise (unless you go into `adpater._entry...` etc. The adapter is also implicitly validating the JSON you pass it as an OPTIMADE resource so there is some useful work being done.

Reply:
True. Okay - I'll add it in as a converter. Perhaps it should be a valid converter for all adapters? So it may be implemented in the parent.

Reply:
I've now added common converters in the base adapter for `json` and `dict`, i.e., one can call `entry.get_json` or `entry.get_dict` and it should work.
For now there is no logic in place to check that `json` and `dict` are not added to `_type_converters`.